File: urbanairship-sdk/src/main/java/com/urbanairship/google/GooglePlayServicesUtilWrapper.java
/* Copyright 2017 Urban Airship and Contributors */

package com.urbanairship.google;

import android.content.Context;
import android.support.annotation.NonNull;

import com.google.android.gms.common.GoogleApiAvailability;


/**
 * Wrapper around Google Play Service Utils.
 */
public class GooglePlayServicesUtilWrapper {


    /**
     * Wraps {@link com.google.android.gms.common.GooglePlayServicesUtil#isUserRecoverableError(int)}.
     */
    public static boolean isUserRecoverableError(int errorCode) {
        return GoogleApiAvailability.getInstance().isUserResolvableError(errorCode);
    }

    /**
     * Wraps {@link com.google.android.gms.common.GooglePlayServicesUtil#isGooglePlayServicesAvailable(android.content.Context)}.
     */
    public static int isGooglePlayServicesAvailable(@NonNull Context context) {
        return GoogleApiAvailability.getInstance().isGooglePlayServicesAvailable(context);
    }
}

File: urbanairship-core/src/main/java/com/urbanairship/google/GooglePlayServicesUtilWrapper.kt
/* Copyright Airship and Contributors */
package com.urbanairship.google

import android.content.Context
import com.google.android.gms.common.GoogleApiAvailability

/**
 * Wrapper around Google Play Service Utils.
 */
public object GooglePlayServicesUtilWrapper {

    /**
     * Wraps [com.google.android.gms.common.GooglePlayServicesUtil.isUserRecoverableError].
     */
    @JvmStatic
    public fun isUserRecoverableError(errorCode: Int): Boolean {
        return GoogleApiAvailability.getInstance().isUserResolvableError(errorCode)
    }

    /**
     * Wraps [com.google.android.gms.common.GooglePlayServicesUtil.isGooglePlayServicesAvailable].
     */
    @JvmStatic
    public fun isGooglePlayServicesAvailable(context: Context): Int {
        return GoogleApiAvailability.getInstance().isGooglePlayServicesAvailable(context)
    }
}
